### DISCUSSION TITLE
Have full, unrestricted access to Thunderbird and your computer

### DISCUSSION THREAD

**Discussion by fperal** [Extended-Thunder/send-later#250] [Q&A] [answered]
Upgrading from a previous version ask to fulfill this permission from thunderbird, sounds scary.

**Comment by jph-sendlater** [comment, marked as answer]
From @jkamens responding to the same question a few days ago:

The old version already had access to "the whole computer," Thunderbird is just being explicit about it now.

The new version of Thunderbird supports "MailExtensions", which are similar to Chrome's "WebExtensions" in that they run inside a restricted sandbox with limited access. Before TB78 there were no such restrictions, and every add-on had pretty much full access.

The new security model is great, but the problem is that the Thunderbird team hasn't yet implemented all of the MailExtension API's necessary for add-ons like Send Later to work. They're working on it, but in the meantime, they've provided add-ons with the ability for part of their code to run in the old, pre-MailExtension environment, so that they can do their jobs until the necessary restricted API's are created.

Send Later falls into this category: it is not yet possible to do what Send Later does without going outside of the MailExtension environment. And since any extension which goes outside of the MailExtension environment has access to "the whole computer," Thunderbird warns about that when you install the add-on.

Again: the level of access Send Later is asking for here is not new, it's just more explicit.

Also, the code in the add-on has been reviewed by the moderators of addons.thunderbird.net and confirmed not to be malicious.
